auth/user_mgt.go

```diff
@@ -0,0 +1,493 @@
+	return c.getUser(ctx, map[string]interface{}{"localId": []string{uid}})
+}
+
+//GetUserByPhone returns the user by phone number
+func (c *Client) GetUserByPhone(ctx context.Context, phone string) (*ExportedUserRecord, error) {
+	return c.getUser(ctx, map[string]interface{}{"phoneNumber": []string{phone}})
+}
+
+//GetUserByEmail returns the user by the email
+func (c *Client) GetUserByEmail(ctx context.Context, email string) (*ExportedUserRecord, error) {
+	return c.getUser(ctx, map[string]interface{}{"email": []string{email}})
+}
+
+// UserIterator is the struct behind the Users Iterator
+// https://github.com/GoogleCloudPlatform/google-cloud-go/wiki/Iterator-Guidelines
+type UserIterator struct {
+	client   *Client
+	ctx      context.Context
+	nextFunc func() error
+	pageInfo *iterator.PageInfo
+	users    []*ExportedUserRecord
+}
+
+// Users returns an iterator over the Users
+func (c *Client) Users(ctx context.Context, startToken string) *UserIterator {
+	it := &UserIterator{
+		ctx:    ctx,
+		client: c,
+	}
+	it.pageInfo, it.nextFunc = iterator.NewPageInfo(
+		it.fetch,
+		func() int { return len(it.users) },
+		func() interface{} { b := it.users; it.users = nil; return b })
+	it.pageInfo.MaxSize = maxResults
+	it.pageInfo.Token = startToken
+	return it
+}
+
+// PageInfo supports pagination. See the google.golang.org/api/iterator package for details.
+// This makes UserIterator comply with the Pageable interface.
+func (it *UserIterator) PageInfo() *iterator.PageInfo { return it.pageInfo }
+
+// Next returns the next result. Its second return value is iterator.Done if
+// there are no more results. Once Next returns iterator.Done, all subsequent
+// calls will return iterator.Done.
+func (it *UserIterator) Next() (*ExportedUserRecord, error) {
+	if err := it.nextFunc(); err != nil {
+		return nil, err
+	}
+	user := it.users[0]
+	it.users = it.users[1:]
+	return user, nil
+}
+
+// SetCustomUserClaims sets the user claims (received as a *map[string]:interface{})
+func (c *Client) SetCustomUserClaims(ctx context.Context, uid string, customClaims *map[string]interface{}) error {
+	if customClaims == nil || *customClaims == nil || len(*customClaims) == 0 {
+		customClaims = &map[string]interface{}{}
+	}
+	ur, err := c.UpdateUser(ctx, uid, &UserParams{CustomClaims: customClaims})
+	if err != nil {
+		return err
+	}
+	if ur.UID == uid {
+		return nil
+	}
+
+	return fmt.Errorf("uid mismatch on returned user")
+}
+
+func (c *Client) getUser(ctx context.Context, params map[string]interface{}) (*ExportedUserRecord, error) {
+	resp, err := c.makeUserRequest(ctx, "getAccountInfo", params)
+	if err != nil {
+		return nil, err
+	}
+	var gur getUserResponse
+	err = json.Unmarshal(resp, &gur)
+	if err != nil {
+		return nil, err
+	}
+	if len(gur.Users) == 0 {
+		return nil, fmt.Errorf("cannot find user %v", params)
+	}
+	return makeExportedUser(gur.Users[0])
+}
+
+func (up *userParams) setDeleteFields() {
```

Comment:
Recommend just dissolving this function, and moving the corresponding logic into UpdateUser()